xds/src/main/java/io/grpc/xds/RouteMatch.java

```diff
@@ -196,6 +229,36 @@ public String toString() {
       this.isInvertedMatch = isInvertedMatch;
     }
 
+    private boolean matchesValue(@Nullable Set<String> values) {
+      if (presentMatch != null) {
+        return (values == null) == presentMatch.equals(isInvertedMatch);
+      }
+      if (values == null) {
+        return false;
+      }
+      boolean baseMatch = false;
+      for (String value : values) {
+        if (exactMatch != null) {
+          baseMatch |= exactMatch.equals(value);
```

Comment:
nit: the following might be easier to follow

```java
for (String value : values) {
  if (exactMatch != null) {
    baseMatch = exactMatch.equals(value);
  } else if {
    ...
  }
  if (baseMatch) {
    break;
  }
}
```
```

Reply:
Sure.